### PR TITLE
Update `TagHelperSampleTest` to avoid failures due to statefulness of the service

### PR DESCRIPTION
**Pull request by dougbu**
- have one test muted at the moment in our CI
- add `Reset()` action to the service, enabling `// Arrange` to bring `Index()` content to a known state
  - add a couple more tests to `TagHelperSampleTest` covering the new action and empty `Index()` list

nits:
- use `ViewData`, not `ViewBag`

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/TagHelperSample.Web.Home.Index.html [hunk 1]
```diff
@@ -49,20 +49,20 @@
             </p>
             <div class="form-group">
                 <label for="z1__Name">Name</label>
-                <input type="text" disabled="disabled" readonly="readonly" id="z1__Name" name="[1].Name" value="Billy" />
+                <input type="text" disabled="disabled" readonly="readonly" id="z1__Name" name="[1].Name" value="Bobby" />
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/TagHelperSample.Web.Home.Index.html]: Previously user `0` and `1` were identical.

**Comment by dougbu** [comment]
Opps, corrected comment to /cc @pranavkm

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TagHelperSampleTest.cs [hunk 2]
```diff
@@ -49,6 +65,34 @@ public TagHelperSampleTest(MvcTestFixture<TagHelperSample.Web.Startup> fixture)
 #endif
         }
 
+        [Fact]
+        public async Task HomeController_Index_ReturnsExpectedContent_AfterReset()
+        {
+            // Arrange
+            var resetResponse = await Client.PostAsync("http://localhost/Home/Reset", content: null);
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TagHelperSampleTest.cs]: Refactor this and next line into a helper method for future tests. The pattern is used a few times in the file now.

**Comment by NTaylorMullen** [comment]
:shipit:

File: samples/TagHelperSample.Web/Controllers/HomeController.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public IActionResult Index()
         // GET: /Home/Create
         public IActionResult Create()
         {
-            ViewBag.Items = _items;
+            ViewData["Items"] = _items;
```
Comment (pranavkm) [inline samples/TagHelperSample.Web/Controllers/HomeController.cs]: Why change this?

**Comment by pranavkm** [comment]
Should the statics be on a singleton service? In which case we wouldn't need to reset.

:watch:

File: samples/TagHelperSample.Web/Controllers/HomeController.cs (hunk 3 above)
Comment (dougbu) [inline samples/TagHelperSample.Web/Controllers/HomeController.cs]: Because we've moved completely away from demonstrating the horrendously poor performance of `ViewBag`. Look at any current ASP.NET Core project template.

Comment (pranavkm) [inline samples/TagHelperSample.Web/Controllers/HomeController.cs]: We should keep `ViewBag` around to for issues such as https://github.com/aspnet/Mvc/issues/3991. No reason to change it here.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TagHelperSampleTest.cs (hunk 2 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TagHelperSampleTest.cs]: Nah, 2 line helpers make the code harder to grok.

Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TagHelperSampleTest.cs]: This is a _sample_, not a functional test.

File: samples/TagHelperSample.Web/Controllers/HomeController.cs (hunk 3 above)
Comment (dougbu) [inline samples/TagHelperSample.Web/Controllers/HomeController.cs]: This is a _sample_, not a functional test site.

**Comment by dougbu** [comment]
> Should the statics be instances on a singleton service?

Wouldn't help because there's only one copy of this site per execution of all tests in the `TagHelperSampleTest` class.

**Comment by dougbu** [comment]
7446cec
